pkg/scheduler/service/cleaner.go

```diff
@@ -61,3 +125,99 @@ func (c *cleaner) purgeReconciliations(transition *ClusterStatusTransition, conf
 		}
 	}
 }
+
+func (c *cleaner) getLatestReconciliations(latestEntitiesCount int, transition *ClusterStatusTransition) ([]*model.ReconciliationEntity, error) {
+	return transition.ReconciliationRepository().GetReconciliations(&reconciliation.Limit{Count: latestEntitiesCount})
+}
+
+func (c *cleaner) dropRecordsOlderThan(t time.Time, transition *ClusterStatusTransition) error {
+	deadline := t.UTC()
+	list, err := c.findReconciliationsOlderThan(deadline, transition)
+	if err != nil {
+		return err
+	}
+
+	c.logger.Infof("[CLEANER] Removing records older than: %s. Records found: %d", deadline.String(), len(list))
+	if len(list) > 0 {
+		c.removeReconciliations(list, transition, func(m *model.ReconciliationEntity) bool { return true })
+	}
+	return err
+}
+
+func (c *cleaner) dropSuccessfulRecordsOlderThan(t time.Time, transition *ClusterStatusTransition) error {
+	deadline := t.UTC()
+	list, err := c.findReconciliationsOlderThan(deadline, transition)
+	if err != nil {
+		return err
+	}
+
+	c.logger.Infof("[CLEANER] Removing successful records older than: %s. Records found: %d (only some must be removed)", deadline.String(), len(list))
+	if len(list) > 0 {
+		c.removeReconciliations(list, transition, func(m *model.ReconciliationEntity) bool { return m.Status.IsFinalStable() })
```

Comment:
Can we give this anonymous function a proper name?

Reply:
Done.